source/redux/icp.js

```diff
@@ -15,4 +19,35 @@ export const walletSlice = createSlice({
 
 export const { setICPPrice } = walletSlice.actions;
 
+export const selectICPPrice = (store) => store.icp.icpPrice;
+
+/**
+ * @description Hook that takes the ICP price from the coingecko API and sets it to the store.
+ * @param {Array.<*>} dependencies Array of dependencies for useEffect
+ * @returns {number} icpPrice
+ */
+export const useICPPrice = (refetch = false, dependencies = []) => {
+  const dispatch = useDispatch();
+  const icpPrice = useSelector(selectICPPrice);
+
+  const fetchICPPrice = async () => {
+    try {
+      const { data } = await getICPPrice();
+
+      dispatch(setICPPrice(data?.['internet-computer']?.usd ?? 1));
+    } catch (error) {
+      // TODO: Handle error
```

Comment:
What are will be our steps to obtain ICP price here in case if `getICPPrice` will fail?
I thought about calling other API to grab it

Reply:
are there any other api's we can call? we could also re-try the call

Reply:
@angarita-dev re-try sounds good
I am just afraid if api will be fully down and we will be strongly rely on it

Reply:
The correct way to do it would be to code some sort of price manager, like a little piece which does the request and if it fails just tries to another provider in the list. Not sure how worth it it is, this works pretty good and has rarely failed. Copying @ChriLnth

Reply:
My suggestion is to leave it as that, unless some problems will pop up
https://status.coingecko.com/ - SLA is pretty good